bitcoin/src/address.rs

```diff
@@ -955,6 +977,11 @@ impl Address {
         let payload = self.payload.inner_prog_as_bytes();
         payload == xonly_pubkey.serialize()
     }
+
+    /// Returns true if the address creates a particular script
```

Comment:
A comment about not allocating would be helpful.

Reply:
added